### PR TITLE
feat(base-cluster/nginx): add possibility to disable ingress

### PR DESCRIPTION
**Pull request by marvinWolff**
Add ability to disable ingress.

### CODE REVIEW

File: charts/base-cluster/templates/monitoring/kube-prometheus-stack/_helpers.yaml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {{- define "base-cluster.monitoring.ingress" -}}
   {{- $ingress := dig .name "ingress" nil .context.Values.monitoring | required (printf "You need to set the `ingress` for %s" .name) -}}
-  {{- if and .context.Values.certManager.email $ingress.enabled (or .context.Values.global.baseDomain $ingress.customDomain) -}}
+  {{- if and .context.Values.certManager.email $ingress.enabled .context.Values.ingress.enabled (or .context.Values.global.baseDomain $ingress.customDomain) -}}
     {{- $host := include (printf "base-cluster.%s.host" .name) .context -}}
 ingress:
-  enabled: true
+  enabled: {{ .context.Values.ingress.enabled }}
```
[the remark this answers is not in the document]
Comment (cwrau) [reply charts/base-cluster/templates/monitoring/kube-prometheus-stack/_helpers.yaml:6]: That's not a supported situation.
If you want to use the included ingress configuration you have to use the included ingress controller.

If you want to provide your own ingress controller you have to create your own ingresses for the built-in resources.

**Comment by cwrau** [comment]
You're missing the `if` in

- [ ] https://github.com/teutonet/teutonet-helm-charts/blob/93b4747ddef340470c27162347971515afb9ab59/charts/base-cluster/templates/global/cluster-ingress.yaml#L1-L0

**Review by cwrau — CHANGES_REQUESTED** [review]

File: charts/base-cluster/templates/monitoring/kube-prometheus-stack/_helpers.yaml (hunk 1 above)
Comment (cwrau) [inline charts/base-cluster/templates/monitoring/kube-prometheus-stack/_helpers.yaml:6]: You don't need to do both, and thinking about it, I think I would prefer the check in the `if`

**Review by cwrau — CHANGES_REQUESTED** [review]

**Review by cwrau — DISMISSED** [review]

**Review by tasches — CHANGES_REQUESTED** [review]
the namespace ingress-nginx should not be created if ingress is disabled

**Comment by marvinWolff** [comment]
> the namespace ingress-nginx should not be created if ingress is disabled

Added :+1:

**Review by cwrau — CHANGES_REQUESTED** [review]
Ah, you're also missing the dashboard in grafana

And the clusterissuer http01 solver

**Review by tasches — CHANGES_REQUESTED** [review]
I guess the helmrepository ingress-nginx and the grafana dashboard can also be excluded if the nginx is disabled.
Also there is a regex left behind for that namespace in alertmanager.additionalAlertRelabelConfigs which can be left as it is but is not necessary if the namespace does not exist.

**Comment by marvinWolff** [comment]
I've now disabled the dashboards and disabled the cluster issuer if the ingress is disabled.

**Review by cwrau — APPROVED** [review]

**Review by tasches — APPROVED** [review]

**Pull request merged**